test/katex-spec.js

```diff
@@ -1591,6 +1591,16 @@ describe("A comment parser", function() {
         expect("% comment 1\n% comment 2\n").toParse();
     });
 
+    it("should parse comments between subscript and superscript", () => {
+        expect("x_3 %comment\n^2").toParseLike`x_3^2`;
+    });
+
+    it("should parse comments in size and color groups", () => {
+        expect("\\kern{1 %kern\nem}").toParse();
+        expect("\\kern1 %kern\nem").toParse();
+        expect("\\color{#f00%red\n}").toParse();
+    });
```

Comment:
We should probably have some tests for parsing `raw` arg types.  It would be cool if we had a test that defines a function that accepts one `raw` arg and one optional `raw` arg then we could test that including `[]` or `{}` in the raw string works as expected.

Reply:
@kevinbarabash Currently, it seems there is no way to define a function cleanly in the tests. I've updated existing test case below.

Reply:
Maybe that's something we can do in the future.  I've opened an issue for it https://github.com/Khan/KaTeX/issues/1740.